### PR TITLE
Added Settings Menu to Levels

### PR DESCRIPTION
**Pull request by spicywheatbread**
New MenuHolder object that holds all menus and enables/disables. Settings menu now accessible in levels.

### CODE REVIEW

File: Assets/Scripts/UI/LevelSelector.cs [hunk 1]
```diff
@@ -8,21 +8,15 @@ public class LevelSelector : MonoBehaviour
...
+    void Awake()
     {
         levelLoader = FindObjectOfType<SceneLoader>();
     }
 
-    public void Return()
-    {
-        levelLoader.LoadMainMenu();
-    }
-
```
[the remark this answers is not in the document]
Comment (spicywheatbread) [reply Assets/Scripts/UI/LevelSelector.cs:25]: There's an OnClick() that sets levelSelector to inactive, and just takes it off the screen. It's not a separate scene

File: Assets/Scripts/UI/MainMenu.cs [hunk 2]
```diff
@@ -9,7 +9,6 @@ public class MainMenu : MonoBehaviour
     [Header("Buttons")]
     public Button playButton;
     public Button loadButton;
-    public Button levelsButton;
```
[the remark this answers is not in the document]
Comment (spicywheatbread) [reply Assets/Scripts/UI/MainMenu.cs:12]: We're not using it so I thought might as well get rid of it

File: Assets/SettingsButton.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+public class SettingsButton : MonoBehaviour
+{
+    [SerializeField] GameObject settingsMenu;
+    [SerializeField] GameObject PauseMenu;
+
+    public void enableMenu() {
+        settingsMenu.SetActive(true);
+        //PauseMenu.GetComponent<PauseMenu>().enableCanvas(false);
+    }
+}
```
[the remark this answers is not in the document]
Comment (spicywheatbread) [reply Assets/SettingsButton.cs:14]: yea sure

File: Assets/UIButton.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    void Start()
+    {
+        audioSys = FindObjectOfType<AudioSystem>();
+    }
+
+    // Update is called once per frame
+    public void buttonSFX() {
+        audioSys.PlaySFX("NEWpageflip");
+    }
+}
```
[the remark this answers is not in the document]
Comment (spicywheatbread) [reply Assets/UIButton.cs:18]: sound sgood

File: Assets/Scripts/UI/SettingsMenu.cs [hunk 5]
```diff
@@ -10,6 +10,8 @@ public class SettingsMenu : MonoBehaviour
     [Header("Audio Mixer")]
     [SerializeField] AudioMixer audioMixer;
     [SerializeField] AudioMixerParamsConfig mixerParams;
+    [SerializeField] GameObject PauseMenu;
```
[the remark this answers is not in the document]
Comment (spicywheatbread) [reply Assets/Scripts/UI/SettingsMenu.cs:13]: maybe at some point it was, and i forgot to take it out. ill just remove it

File: Assets/Scripts/UI/SettingsMenu.cs [hunk 6]
```diff
@@ -10,6 +10,8 @@ public class SettingsMenu : MonoBehaviour
...
+    [SerializeField] GameObject Settings;
```
[the remark this answers is not in the document]
Comment (spicywheatbread) [reply Assets/Scripts/UI/SettingsMenu.cs:14]: yea, it does, so ill do that.

**Review by jalenng — APPROVED** [review]

**Pull request merged**